torchinductor/ir.py

```diff
@@ -596,30 +606,32 @@ def fn(index):
         return TensorBox.create(
             Reduction(
                 device,
-                dtype,
+                dst_dtype,
                 inner_fn,
                 ranges,
                 reduction_ranges,
                 reduction_type,
+                src_dtype,
             )
         )
 
     @staticmethod
     def default_value(reduction_type, dtype):
+        if reduction_type in {"max", "argmax"}:
+            return float("-inf") if is_float_dtype(dtype) else -(2**31)
```

Comment:
This should depend on the dtype? E.g. for max/argmax on uint8 tensor this will overflow, for long tensor potentially produce wrong results?

Reply:
You are right. I initially used 2**63, but it is better to write a util function for this (let me update in another PR), or is there any existing python util for it?

Reply:
There's `torch.iinfo(dtype).max`